Historical info on Jim, acknowledgement and thanks

jaustinhughey opened the pull request:
This is just a friendly (albeit somber) pull request to acknowledge Jim Weirich in a more salient, respectful way. The original copyright statement under "other stuff", regardless of who put it there, now seems almost disrespectful, as if he's an afterthought and nothing more.

It's important that we acknowledge and thank innovators for their work, even in death; such ongoing acknowledgement shows veterans and newcomers alike that building great things leaves a lasting legacy; seeing that legacy can inspire others to build *their own* legacy by contributing to the community.

More functionally, newcomers to Ruby might see lots of references to Jim and jimweirich/rake as the repository location (e.g. books, old articles/blog posts, etc.), so this may clear up some confusion.

Merely a friendly suggestion; please discuss or ask questions if you have them. Thanks!

rdlugosz commented:
I think this a good idea. As we get further down the road it would help
introduce new programmers to Jim's legacy.

I would include a link to Jim's Wikipedia entry:
https://en.m.wikipedia.org/wiki/Jim_Weirich

I'll look this over some more when not on my phone and maybe offer some
additional changes. Thanks for contributing this!

-Ryan

On Monday, September 7, 2015, J. Austin Hughey <notifications@github.com>
wrote:

> This is just a friendly (albeit somber) pull request to acknowledge Jim
> Weirich in a more salient, respectful way. The original copyright statement
> under "other stuff", regardless of who put it there, now seems almost
> disrespectful, as if he's an afterthought and nothing more.
>
> It's important that we acknowledge and thank innovators for their work,
> even in death; such ongoing acknowledgement shows veterans and newcomers
> alike that building great things leaves a lasting legacy; seeing that
> legacy can inspire others to build *their own* legacy by contributing to
> the community.
>
> More functionally, newcomers to Ruby might see lots of references to Jim
> and jimweirich/rake as the repository location (e.g. books, old
> articles/blog posts, etc.), so this may clear up some confusion.
>
> Merely a friendly suggestion; please discuss or ask questions if you have
> them. Thanks!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/ruby/rake/pull/66
> Commit Summary
>
>    - acknowledge Jim in a more salient way
>
> File Changes
>
>    - *M* README.rdoc <https://github.com/ruby/rake/pull/66/files#diff-0>
>    (13)
>
> Patch Links:
>
>    - https://github.com/ruby/rake/pull/66.patch
>    - https://github.com/ruby/rake/pull/66.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ruby/rake/pull/66>.
>

drbrain commented:
I don't see any issue with this.

Note that it was [Jim himself](https://github.com/ruby/rake/blame/master/README.rdoc#L124) who wrote the "Other stuff" section originally and the subsequent commits were minor cleanups.

jaustinhughey commented:
@drbrain Yeah, when I originally saw the "other stuff" item I was like, "what, seriously? How rude!" but after looking it up a bit more I realized the same thing. He wasn't an egomaniac, that's for sure :)

@rdlugosz Good idea on the wikipedia entry; I've updated the pull request with a link to said article and also fixed some syntax issues I just noticed - I'm used to Markdown, not rdoc syntax :)

Please feel free to ask additional questions if you have them, or merge as you see fit. Thanks!

hsbt commented:
@jaustinhughey Thank you! I merged your suggestion.

jaustinhughey commented:
Thank you very much, @hsbt! It's sincerely appreciated.

pvdb commented:
@jahio - thanks for this PR :pray: but just wondering why you decided to link to the [git tree @ `336559f`](https://github.com/jimweirich/rake/tree/336559f28f55bce418e2ebcc0a57548dcbac4025) instead of to the [actual commit](https://github.com/jimweirich/rake/commit/336559f28f55bce418e2ebcc0a57548dcbac4025) _(which is also where people have also left their condolences over the years)_?

If you agree linking to the commit makes sense, then I'm happy to submit a PR along the lines of:

```diff
@@ -148,7 +148,7 @@ February 2014. This repository was originally hosted at
 with his passing, has been moved to {ruby/rake}[https://github.com/ruby/rake].
 
 You can view Jim's last commit here:
-https://github.com/jimweirich/rake/tree/336559f28f55bce418e2ebcc0a57548dcbac4025
+https://github.com/jimweirich/rake/commit/336559f28f55bce418e2ebcc0a57548dcbac4025
 
 You can {read more about Jim}[https://en.wikipedia.org/wiki/Jim_Weirich] at Wikipedia.
```

@hsbt - any thoughts?

hsbt commented:
@pvdb I'm +1 to your suggestion. Can you make a PR?

jahio commented:
> @jahio - thanks for this PR 🙏 but just wondering why you decided to link to [the git tree @ `336559f`](https://github.com/jimweirich/rake/tree/336559f28f55bce418e2ebcc0a57548dcbac4025) instead of to [the actual commit](https://github.com/jimweirich/rake/commit/336559f28f55bce418e2ebcc0a57548dcbac4025) _(which is also where people have left their condolences over the years)_?
> 
> If you agree linking to the commit makes sense, then I'm happy to submit a PR along the lines of:
> 
> ```diff
> @@ -148,7 +148,7 @@ February 2014. This repository was originally hosted at
>  with his passing, has been moved to {ruby/rake}[https://github.com/ruby/rake].
>  
>  You can view Jim's last commit here:
> -https://github.com/jimweirich/rake/tree/336559f28f55bce418e2ebcc0a57548dcbac4025
> +https://github.com/jimweirich/rake/commit/336559f28f55bce418e2ebcc0a57548dcbac4025
>  
>  You can {read more about Jim}[https://en.wikipedia.org/wiki/Jim_Weirich] at Wikipedia.
> ```
> 
> @hsbt - any thoughts?

If you look at the date of my PR, it was submitted in 2015. The oldest comment on the commit you referenced came nearly 4 years later. At the time I created this pull request, those comments had yet to exist.

The reason I linked to the tree, not the specific commit, is because _from_ the tree, one could navigate _to_ the commit, as well as see the state of the readme as it existed at that point in time, comparing it for themselves against the state of the official project readme as it exists in that person's present, no matter what year it happens to be. It's now nine years hence; many things could have changed since then in terms of the scope of the project, etc. Things we can't predict nor should we take for granted could still change in the years to come (and of course I thought of this back then as well).

Another element of my thinking at the time, especially for people new to Ruby (or for the non-technical folks out there), was to ease them into things by showing them something analogous to what they just read - a readme - not a literal code commit, when they wind up at the destination referenced by that link. Thinking it through from their perspective.

This is, in the grand scheme of things, all minor stuff. The important thing is acknowledging Jim, so any way that gets that done is fine with me.

I hope this answers your question, @pvdb. I just didn't want to leave you hanging, with a question unanswered. Do let me know if there's anything else I can help clarify.

pvdb commented:
Thanks for the comment, @jahio, and apologies for the tardy reply ... I have been AFK for 2+ weeks, so just picking it up now. :sweat_smile:

Your rationale and very detailed clarification make a lot of sense: instead of linking to Jim's last contribution, I can see why you linked to the state of `rake` as he left it, so to speak.

Based on your reply, I should've held off with #593 so I've just reverted it in #600 for @hsbt 's consideration :pray:

rdlugosz commented:
I'd like to think Jim would smile if he saw the care that's gone into preserving his legacy... and I know he'd get a good laugh out of debating the semantics of link targets :)

Cheers all, rip Jim.
